src/traits/mod.rs

```diff
@@ -65,6 +65,14 @@ pub trait DoughnutApi {
     }
 }
 
+pub trait Signing {
```

Comment:
@holygits What if here we have only one single entry for `fn sign(...args)`, and encapsulate various kind of signing methods, only if signing function always takes `publickey, secretkey, payload` as arguments?

cc @alexsednz @KarishmaBothara

Reply:
Umm... maybe it would be better to expose determined api interfaces like
```js
sign_sr25519(pubkey, scrtkey, payload);
// or
sign_ed25519(pubkey, scrtkey, payload);
```
rather than
```js
sign(pubkey, scrtkey, payload, 'sr25519');
```

Reply:
don't need `payload` arg anymore, can use the doughnut directly `self.payload()`.
not sure we need to pass `public_key` at all, `from_seed_*` functions will be enough to create the keypair